From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Fri, 6 May 2016 03:07:40 +0200
Subject: [PATCH] deploy.sh: Remove check for root (rm umask 0000)

Patch(es) by Josep, signed by Alex only for upstreaming.

[Alexandru Avadanii]
Squashed two atomic commits into one change.

Signed-off-by: Josep Puigdemont <josep.puigdemont@enea.com>
Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 ci/deploy.sh                        | 11 -----------
 deploy/deploy-config.py             |  1 -
 deploy/deploy.py                    |  2 --
 deploy/environments/virtual_fuel.py |  2 --
 4 files changed, 16 deletions(-)

diff --git a/ci/deploy.sh b/ci/deploy.sh
index 8411714..081806c 100755
--- a/ci/deploy.sh
+++ b/ci/deploy.sh
@@ -84,9 +84,6 @@ Input parameters to the build script is:
 -i .iso image to be deployed (needs to be provided in a URI
    style, it can be a local resource: file:// or a remote resource http(s)://)

-NOTE: Root priviledges are needed for this script to run
-
-
 Examples:
 sudo `basename $0` -b file:///home/jenkins/lab-config -l lf -p pod1 -s ha_odl-l3_heat_ceilometer -i file:///home/jenkins/myiso.iso
 EOF
@@ -212,11 +209,6 @@ do
     esac
 done

-if [[ $EUID -ne 0 ]]; then
-    echo "This script must be run as root" 1>&2
-    exit 1
-fi
-
 if [ -z $BASE_CONFIG_URI ] || [ -z $TARGET_LAB ] || \
    [ -z $TARGET_POD ] || [ -z $DEPLOY_SCENARIO ] || \
    [ -z $ISO ]; then
@@ -231,9 +223,6 @@ fi
 # Enable the automatic exit trap
 trap do_exit SIGINT SIGTERM EXIT

-# Set no restrictive umask so that Jenkins can removeeee any residuals
-umask 0000
-
 clean

 pushd ${DEPLOY_DIR} > /dev/null
diff --git a/deploy/deploy-config.py b/deploy/deploy-config.py
index 2a09aa3..02debe2 100644
--- a/deploy/deploy-config.py
+++ b/deploy/deploy-config.py
@@ -41,7 +41,6 @@ from common import (
     check_file_exists,
     create_dir_if_not_exists,
     delete,
-    check_if_root,
     ArgParser,
 )

diff --git a/deploy/deploy.py b/deploy/deploy.py
index fe213e5..08702d2 100755
--- a/deploy/deploy.py
+++ b/deploy/deploy.py
@@ -34,7 +34,6 @@ from common import (
     check_dir_exists,
     create_dir_if_not_exists,
     delete,
-    check_if_root,
     ArgParser,
 )

@@ -252,7 +251,6 @@ class AutoDeploy(object):
         return 0

     def run(self):
-        check_if_root()
         if self.cleanup_only:
             self.cleanup_execution_environment()
         else:
diff --git a/deploy/environments/virtual_fuel.py b/deploy/environments/virtual_fuel.py
index b1a76e4..4ff68f6 100644
--- a/deploy/environments/virtual_fuel.py
+++ b/deploy/environments/virtual_fuel.py
@@ -18,7 +18,6 @@ import time
 from common import (
     exec_cmd,
     check_file_exists,
-    check_if_root,
     delete,
     log,
 )
@@ -135,7 +134,6 @@ class VirtualFuel(ExecutionEnvironment):
                        vm_definition_overwrite)

     def setup_environment(self):
-        check_if_root()
         self.cleanup_environment()
         self.create_vm()

